### PR TITLE
Replaced the Popen commands in check_cert_key to use OpenSSL

### LINKED ISSUE #313: Error when running sender and receiver agents on a MAC
When running the sender and receiver daemon (using `python bin/receiver.py -c ~/DCIG/ssm/conf/receiver.cfg` and `python bin/sender.py -c ~/DCIG/ssm/conf/sender.cfg`)

An error response is returned, this could be because I am using a MAC with OpenSSL version 3.2.1 and pyOpenSSL          24.1.0

Sender log message:
```
2024-04-04 13:15:54,368 - ssmsend - INFO - ========================================
2024-04-04 13:15:54,369 - ssmsend - INFO - Starting sending SSM version 3.4.0.
2024-04-04 13:15:54,369 - ssmsend - INFO - Setting up SSM with protocol: AMS
2024-04-04 13:15:54,369 - ssmsend - INFO - No server certificate supplied.  Will not encrypt messages.
2024-04-04 13:15:54,372 - ssmsend - INFO - Messages will be signed using /C=XX/ST=StateName/L=CityName/O=CompanyName/OU=CompanySectionName/CN=CommonNameOrHostname
2024-04-04 13:15:54,391 - ssm.crypto - ERROR - Warning: Reading certificate from stdin since no -in or -new option is given

SSM failed to complete successfully.  See log file for details.
2024-04-04 13:15:54,391 - ssmsend - ERROR - SSM failed to complete successfully: Cert and key don't match.
2024-04-04 13:15:54,392 - ssmsend - INFO - SSM has shut down.
2024-04-04 13:15:54,392 - ssmsend - INFO - ========================================
Traceback (most recent call last):
  File "~/DCIG/ssm/bin/sender.py", line 79, in <module>
    main()
  File "~/DCIG/ssm/bin/sender.py", line 75, in main
    ssm.agents.run_sender(protocol, brokers, project, token, cp, log)
  File "~/DCIG/ssm/ssm/agents.py", line 264, in run_sender
    if sender_failed:
UnboundLocalError: local variable 'sender_failed' referenced before assignment
```

Receiver log message:
```
2024-04-04 11:21:13,032 - ssmreceive - INFO - ========================================
2024-04-04 11:21:13,033 - ssmreceive - INFO - Starting receiving SSM version 3.4.0.
2024-04-04 11:21:13,033 - ssmreceive - INFO - Setting up SSM with protocol: AMS
2024-04-04 11:21:13,033 - ssmreceive - INFO - The SSM will run as a daemon.
2024-04-04 11:21:13,069 - ssm.crypto - ERROR - Warning: Reading certificate from stdin since no -in or -new option is given

2024-04-04 11:21:13,069 - ssmreceive - CRITICAL - Failed to initialise SSM: Cert and key don't match.
2024-04-04 11:21:13,069 - ssmreceive - INFO - ========================================
```

in both it has this `Warning: Reading certificate from stdin since no -in or -new option is given`

This is the function that is being ran:
https://github.com/apel/ssm/blob/f43b436ff08c32232e0318947e547298d1769779/ssm/crypto.py#L51

### PR DESCRIPTION
**Pull request by Will-Cross1**
Resolves #313

Solves original problem of a warning returned by Popen as an error and causing the function to return False

Improves the variable names from pubkey1 and pubkey2 to help show where the key came from

### CODE REVIEW

File: ssm/crypto.py [hunk 1]
```diff
@@ -64,23 +64,36 @@ def check_cert_key(certpath, keypath):
...
+        certificate = OpenSSL.crypto.load_certificate(
+            OpenSSL.crypto.FILETYPE_PEM, cert
+        )
+        crypto_public_key = certificate.get_pubkey()
+        public_key_bytes = OpenSSL.crypto.dump_publickey(
+            OpenSSL.crypto.FILETYPE_PEM, crypto_public_key
+        )
 
-    if error != '':
+        certificate_public_key = public_key_bytes.decode("utf-8")
```
Comment (tofu-rocketry) [inline ssm/crypto.py:76]: This seems like an unnecessary step if we're doing this to both keys.

File: ssm/crypto.py [hunk 2]
```diff
@@ -64,23 +64,36 @@ def check_cert_key(certpath, keypath):
...
+
+    except Exception as error:
```
Comment (tofu-rocketry) [inline ssm/crypto.py:78]: Not a fan of base `Exception` being caught, though not sure of all the exceptions that could be generated by this.

**Review by tofu-rocketry — CHANGES_REQUESTED** [review]
The docstring for the method could do with updating as it still refers to using "openssl directly", when it's now done via pyopenssl.

File: ssm/crypto.py (hunk 1 above)
Comment (Will-Cross1) [reply ssm/crypto.py:76]: resolved in next commit

File: ssm/crypto.py (hunk 2 above)
Comment (Will-Cross1) [reply ssm/crypto.py:78]: Resolved in next commit

**Pull request merged**